src/functions-and-arrays.js

```diff
@@ -26,16 +48,69 @@ function sum() {}
 // Level 1: Array of numbers
 const numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
 
-function averageNumbers() {}
+function averageNumbers(array) {
+  
+  if (array.length === 0) {
+    return null;
+  } else {
+    const average = sumNumbers(array) / array.length;
+  return average;
+  }
+ 
+}
 
+averageNumbers(numbers);
 
-// Level 2: Array of strings
+// Level 4.2: Array of strings
 const wordsArr = ['seat', 'correspond', 'linen', 'motif', 'hole', 'smell', 'smart', 'chaos', 'fuel', 'palace'];
 
-function averageWordLength() { }
+function averageWordLength(arr) {
+
+  if (array.length === 0) {
+    return null;
+  } else { 
+    let sum = 0;
+    let average = 0;
+    for(let i = 0; i < arr.length; i++) {
+    let number = arr[i].length;
+    sum += (number.length);
+    }
+    average = sum / arr.length;
+    return average;
+  }
+}
+
+averageWordLength(wordsArr);
+
+// Bonus - Iteration #4.3
+
+function avg(arrayOfWords) {
+  
+  if (arrayOfWords.length === 0) {
+    return null;
+  } else {
+
+    const average = sumNumbers(array) / array.length;
+  return average;
+  }
+ 
+}
+
+
+function avg(arrayOfWords) {
+  let sum = 0;
+
+  arrayOfWords.forEach(item => {
+    sum += item;
```

Comment:
this wouldn't work. it just adds the current word to sum. not the length. also you have this function avg. defined twice. maybe rename or delete/comment out one of them.